internal/connect/command_offset_describe.go

```diff
@@ -0,0 +1,180 @@
+	Id   string `human:"Id"`
+	Name string `human:"Name"`
+}
+
+type offsetsDescribeOut struct {
+	Partition string `human:"Partition"`
+	Offset    string `human:"Offset"`
+}
+
+type metadataDescribeOut struct {
+	ObservedAt string `human:"Observed At"`
+}
+
+func (c *offsetCommand) newDescribeCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:               "get <id>",
+		Short:             "Get connector offsets",
+		Args:              cobra.ExactArgs(1),
+		RunE:              c.getOffset,
+		ValidArgsFunction: pcmd.NewValidArgsFunction(c.validArgs),
+		Annotations:       map[string]string{pcmd.RunRequirement: pcmd.RequireNonAPIKeyCloudLogin},
+		Example: examples.BuildExampleString(
+			examples.Example{
+				Text: "Get connector offsets for a lccId in the current or specified Kafka cluster context.",
+				Code: "confluent connect offset get lcc-123456",
+			},
+			examples.Example{
+				Code: "confluent connect offset get lcc-123456 --cluster lkc-123456",
+			},
+		),
+	}
+
+	pcmd.AddClusterFlag(cmd, c.AuthenticatedCLICommand)
+	pcmd.AddContextFlag(cmd, c.CLICommand)
+	pcmd.AddEnvironmentFlag(cmd, c.AuthenticatedCLICommand)
+	pcmd.AddOutputFlag(cmd)
+
+	return cmd
+}
+
+func (c *offsetCommand) getOffset(cmd *cobra.Command, args []string) error {
+	kafkaCluster, err := kafka.GetClusterForCommand(c.V2Client, c.Context)
+	if err != nil {
+		return err
+	}
+
+	environmentId, err := c.Context.EnvironmentId()
+	if err != nil {
+		return err
+	}
+
+	connectorIdToName, err := c.mapConnectorIdToName(environmentId, kafkaCluster.ID)
+	if err != nil {
+		return err
+	}
+
+	offsets, err := c.V2Client.GetConnectorOffset(connectorIdToName[args[0]], environmentId, kafkaCluster.ID)
+	if err != nil {
+		return err
+	}
+
+	if output.GetFormat(cmd) == output.Human {
+		return printHumanDescribeOffset(cmd, offsets, args[0], connectorIdToName[args[0]])
+	}
+
+	return printSerializedDescribeOffsets(cmd, offsets, args[0], connectorIdToName[args[0]])
+}
+
+func (c *offsetCommand) mapConnectorIdToName(environmentId, kafkaClusterId string) (map[string]string, error) {
+	// NOTE: Do NOT replace this with `V2Client.GetConnectorExpansionById` calls; that function itself calls `V2Client.ListConnectorsWithExpansions`
+	connectors, err := c.V2Client.ListConnectorsWithExpansions(environmentId, kafkaClusterId, "id,info,status")
+	if err != nil {
+		return nil, err
+	}
+
+	connectorIdToName := make(map[string]string)
+	for _, connector := range connectors {
+		connectorIdToName[connector.Id.GetId()] = connector.Info.GetName()
+	}
+
+	return connectorIdToName, nil
+}
+
+func printHumanDescribeOffset(cmd *cobra.Command, offsets connectv1.ConnectV1ConnectorOffsets, id string, name string) error {
+	output.Println(false, "Connector Details")
+	table := output.NewTable(cmd)
+	table.Add(&ConnectDescribeOut{
+		Name: name,
+		Id:   id,
+	})
+	if err := table.Print(); err != nil {
+		return err
+	}
+	output.Println(false, "")
+	output.Println(false, "")
```

Comment:
As of two years ago, we no longer recommend printing multiple tables in a single command. It makes the `--output json` flag unusable. Can we split this into multiple commands or reconsider the formatting?

Reply:
@brianstrauch I have tested with -o json and -o yaml and it is working fine. 
json ouput ```{
  "id": "lcc-123",
  "name": "az-connector",
  "offsets": [
    {
      "offset": {
        "event": 2,
        "file": "mysql-bin.000600",
        "pos": 2001,
        "row": 1,
        "server_id": 1,
        "transaction_id": null,
        "ts_sec": 1711788870
      },
      "partition": {
        "server": "dbzv2"
      }
    }
  ],
  "metadata": {
    "observed_at": "2024-04-02T08:23:33.000000123Z"
  }
}```

Reply:
@brianstrauch considering the json structure above , i dont think we can make this whole thing in one table/list. Will it be okay if its one table  and one list?

Reply:
We're already printing the table in https://docs.confluent.io/confluent-cli/current/command-reference/connect/cluster/confluent_connect_cluster_describe.html, so let's remove it from this command.

Reply:
Does an offset have an ID? Can we support `confluent connect offset describe <id>` or are we forced to print all of the offsets?

Reply:
no, there is no id, we need to print the whole offset object, its actually and array of map with 2 keys, "offset" and "partition".

Reply:
making it a list made sense with offset and partition as column, but the other fields dont fit that format.